torchmetrics/detection/map.py

```diff
@@ -160,22 +147,31 @@ class MAP(Metric):
     <https://github.com/PyTorchLightning/metrics/blob/master/tm_examples/detection_map.py>`_
 
     .. note::
-        This metric is a wrapper for the
+        This metric is following the mAP implementation of
         `pycocotools <https://github.com/cocodataset/cocoapi/tree/master/PythonAPI/pycocotools>`_,
-        which is a standard implementation for the mAP metric for object detection. Using this metric
-        therefore requires you to have `pycocotools` installed. Please install with ``pip install pycocotools`` or
-        ``pip install torchmetrics[detection]``.
+        , a standard implementation for the mAP metric for object detection.
 
     .. note::
         This metric requires you to have `torchvision` version 0.8.0 or newer installed (with corresponding
         version 1.7.0 of torch or newer). Please install with ``pip install torchvision`` or
         ``pip install torchmetrics[detection]``.
 
-    .. note::
-        As the pycocotools library cannot deal with tensors directly, all results have to be transfered
-        to the CPU, this might have an performance impact on your training.
-
     Args:
+        box_format:
+            Input format of given boxes. Supported formats are [‘xyxy’, ‘xywh’, ‘cxcywh’].
+            default: ``xyxy``
+        iou_thresholds:
+            IoU thresholds for evaluation. If set to `None` it corresponds to the stepped range `[0.5,...,0.95]`
+            with step `0.05`. Else provide a list of floats.
+            default: ``None``
+        rec_thresholds:
+            Recall thresholds for evaluation. If set to `None` it corresponds to the stepped range `[0,...,1]`
+            with step `0.01`. Else provide a list of floats.
+            default: ``None``
+        max_detection_thresholds:
+            Thresholds on max detections per image. If set to `None` will use thresholds `[1, 10, 100]`.
+            Else please provide a list of ints.
+            default: ``None``
```

Comment:
```suggestion
        box_format:
            Input format of given boxes. Supported formats are [‘xyxy’, ‘xywh’, ‘cxcywh’].
        iou_thresholds:
            IoU thresholds for evaluation. If set to `None` it corresponds to the stepped range `[0.5,...,0.95]`
            with step `0.05`. Else provide a list of floats.
        rec_thresholds:
            Recall thresholds for evaluation. If set to `None` it corresponds to the stepped range `[0,...,1]`
            with step `0.01`. Else provide a list of floats.
        max_detection_thresholds:
            Thresholds on max detections per image. If set to `None` will use thresholds `[1, 10, 100]`.
            Else please provide a list of ints.
```
all defaults are automatically taken from the function/method signature